File: bindings/kotlin/src/test/kotlin/SmokeTest.kt
import uniffi.amber_core.CaptureException
import uniffi.amber_core.OutputFormat
import uniffi.amber_core.capture
import uniffi.amber_core.captureMarkdown
import uniffi.amber_core.save
import uniffi.amber_core.snapshot
import kotlin.test.Test
import kotlin.test.assertContains
import kotlin.test.assertTrue
import kotlin.test.assertFailsWith

class SmokeTest {
    // A data: URL keeps the test self-contained while still exercising the
    // real capture pipeline through the bundled engine.
    private val url =
        "data:text/html,<html><body><h1>Smoke</h1><p>hello</p></body></html>"

    @Test
    fun markdown() {
        val md = captureMarkdown(url)
        assertContains(md, "Smoke")
    }

    @Test
    fun binaryFormats() {
        val pdf = capture(url, OutputFormat.PDF)
        assertTrue(pdf.size > 4)
        assertTrue(pdf.copyOfRange(0, 4).contentEquals("%PDF".toByteArray()))

        val png = capture(url, OutputFormat.SCREENSHOT)
        assertTrue(png.copyOfRange(0, 4).contentEquals(byteArrayOf(0x89.toByte(), 0x50, 0x4E, 0x47)))
    }

    @Test
    fun saveToFile() {
        val dir = System.getProperty("java.io.tmpdir") + "/amber-kt-smoke"
        val path = save(url, OutputFormat.HTML, dir, "page")
        assertTrue(path.endsWith("page.html"))
        assertTrue(java.io.File(path).exists())
    }

    @Test
    fun badUrlThrows() {
        assertFailsWith<CaptureException.Failed> { captureMarkdown("not a url") }
    }

    @Test
    fun snapshotRendersManyFromOneCapture() {
        // One capture, many formats (Plans.md 10.1/10.3); Snapshot is AutoCloseable.
        snapshot(url, listOf(OutputFormat.MARKDOWN, OutputFormat.PDF)).use { snap ->
            assertContains(snap.markdown(), "Smoke")

            val pdf = snap.render(OutputFormat.PDF)
            assertTrue(pdf.copyOfRange(0, 4).contentEquals("%PDF".toByteArray()))

            val dir = System.getProperty("java.io.tmpdir") + "/amber-kt-smoke"
            val path = snap.save(OutputFormat.READABLE, dir, "snap")
            assertTrue(path.endsWith("snap.txt"))
            assertTrue(java.io.File(path).exists())
        }
    }

    @Test
    fun snapshotBadUrlThrows() {
        assertFailsWith<CaptureException.Failed> {
            snapshot("not a url", listOf(OutputFormat.MARKDOWN))
        }
    }
}

File: bindings/dart/test/smoke_test.dart
// Smoke test for the Dart binding. Run `bindings/dart/generate.sh` first, then:
//   cd bindings/dart && dart pub get && dart test
// A data: URL keeps it self-contained; PDF/screenshot drive a real browser, so
// set AMBER_CHROMIUM_PATH (or let the pinned Chrome for Testing download once).
import 'dart:typed_data';

import 'package:amber_html/amber.dart';
import 'package:test/test.dart';

const url =
    'data:text/html,<html><body><h1>Smoke</h1><p>hello</p></body></html>';

void main() {
  test('markdown contains content', () {
    expect(captureMarkdown(url), contains('Smoke'));
  });

  test('capture returns binary formats', () {
    final pdf = capture(url, Format.pdf);
    expect(pdf.sublist(0, 4), equals(Uint8List.fromList('%PDF'.codeUnits)));

    final png = capture(url, Format.screenshot);
    expect(png.sublist(0, 4), equals(Uint8List.fromList([0x89, 0x50, 0x4E, 0x47])));
  });

  test('snapshot renders many from one capture', () {
    final snap = Snapshot.capture(url, [Format.markdown, Format.pdf]);
    expect(snap.markdown(), contains('Smoke'));
    expect(snap.render(Format.pdf).sublist(0, 4),
        equals(Uint8List.fromList('%PDF'.codeUnits)));
    snap.close();
  });

  test('bad URL throws', () {
    expect(() => captureMarkdown('not a url'), throwsA(isA<CaptureException>()));
    expect(() => Snapshot.capture('not a url', [Format.markdown]),
        throwsA(isA<CaptureException>()));
  });
}
